docs/maintaining.md

```diff
@@ -92,3 +92,118 @@ made to always install the latest released version.
 
 Update the release branch by fast-forwarding it to the appropriate SHA (matching
 the latest tag) and pushing it back upstream.
+
+## `grafana/prometheus` Maintainence
+
+Grafana Labs is using the Agent for their internal monitoring and want to take
+advantage of the Agent to proof-of-concept additions to Prometheus before they
+get moved upstream. A `grafana/prometheus` repository is maintained by Grafana
+Labs where all non-trivial changes will go first. Doing this allows for getting
+Cortex-specific changes moving along faster and providing strong evidence
+towards its usefulness and correctness when it becomes part of an upstream PR.
+
+We are commiting ourselves to doing the following:
+
+1. Always use a recent Prometheus release: the Agent will always vendor a
```

Comment:
We should clarify that we're vendoring a branch that's based on the latest/a recent release.